pkg/config/settings.go

```diff
@@ -57,6 +57,16 @@ func LoadDefaultsOfIBMBlockCSI() error {
 		return fmt.Errorf("environment variable %q was not set", EnvNameCrYaml)
 	}
 
+	err := appendDefaultsFromCrFileToDefaultCrObject(crYamlPath)
+	if err != nil {
+		return err
+	}
+	setDefaultSidecarImageByName()
```

Comment:
is this necessary for this PR?
if not, let's not include it here

Reply:
it's not a big diff, it's a code that I used and thought I can refactor a little bit, what is the difference between this code and the [code](https://github.com/IBM/ibm-block-csi-operator/blob/04618536287d1a5074c177e7a93e8cb56a598552/controllers/ibmblockcsi_controller.go#L216-L218) that you wanted me to remove, it wasn't also related to the PR

Reply:
generally it's ok to make minor non-related improvements, if the PR is small/quick enough, but I don't feel this is the case.
I'm trying to reduce as much as possible (code, comments, etc...).
the change you linked:

1. reduces code (which IMO is easier to review + more important)
2. was at an earlier time of the PR, so we had more time
3. meant to avoid a potential change which IMO would have been worse

but if you prefer to undo the linked change as well, it won't affect my approval of this PR